LogMonitor/src/LogMonitor/EtwMonitor.cpp

```diff
@@ -678,8 +678,23 @@ EtwMonitor::OnRecordEvent(
 
         if (ERROR_SUCCESS != status)
         {
+            // Access the ProviderId from the EVENT_HEADER field
+            GUID providerId = EventRecord->EventHeader.ProviderId;
+
+            LPOLESTR clsidString;
+            if (StringFromCLSID(providerId, &clsidString) != S_OK)
+            {
+                logWriter.TraceError(
+                    Utility::FormatString(L"Failed to convert GUID to string").c_str()

```

Comment:
This failure occurs because the GUID is not a valid one? Might be nice to provide some more context for the user.

Reply:
At this state in the program execution, we've done the check for valid GUID

From the docs for [StringFromCLSID](https://learn.microsoft.com/en-us/windows/win32/api/combaseapi/nf-combaseapi-stringfromclsid) we can only get to this branch when the value returned is E_OUTOFMEMORY.  We can therefore provide this detailed instead of having a vague message that the conversion failed.